core/lib/types/src/fee_model.rs

```diff
@@ -282,7 +282,7 @@ impl FeeParamsV2 {
     fn convert_to_base_token(&self, price_in_wei: u64) -> u64 {
         let conversion_ratio = BigDecimal::from(self.conversion_ratio.numerator.get())
             / BigDecimal::from(self.conversion_ratio.denominator.get());
-        let converted_price_bd = BigDecimal::from(price_in_wei) * conversion_ratio;
+        let converted_price_bd = BigDecimal::from(price_in_wei) / conversion_ratio;
```

Comment:
The fetched ratio [seems to be](https://github.com/matter-labs/zksync-era/blob/6146fce18673df89485d96c5b38a00d6404aa4b2/core/lib/external_price_api/src/coingecko_api.rs#L80) `token/eth`. So if I understand things correctly, before the unit of calculation was `eth * token / eth = token`, and now it's `eth / (token / eth) = eth * eth / token = eth^2 / token`. Am I wrong?

Reply:
This understanding as well as the existing code are both correct. A simple way to understand this is: the denomination we need stay with is `baseToken` so we need to "cancel out" the ETH denomination.

```
conversion_ratio = numerator(in baseToken)/denominator(in ETH)
gasPrice(in baseToken) = gasPrice(in ETH) * conversion_ratio
```

@ilitteri does it make sense to you?

Reply:
I've followed this reasoning:
1 usdt &rarr; 1 usd
1 eth &rarr; 3500usd

If i want to have how many `eths` do i need to buy 1 usdt:
$$\text{Price of base token[eth]} = \frac{1[usd] \times 1[eth]}{3500[usd]} = 0.0002857 [eth]$$ 
Which is roughly the value obtained from `coingecko`:
- https://docs.coingecko.com/v3.0.1/reference/simple-token-price

```json
{
  "0xdac17f958d2ee523a2206206994597c13d831ec7": {
    "eth": 0.0002857
  }
}
```

Now, if the gasprice is in `eth`:
 $$\text{gasprice[usdt]} = \frac{\text{gasprice[eth]}}{\text{Price of base token[eth}]}$$